### PR TITLE
support UUID #3909

### PR DESCRIPTION
**Pull request by mpisanko**
Elixir test now also works

### CODE REVIEW

File: lang/elixr/test/elixr_test.exs [hunk 1]
```diff
@@ -70,19 +70,28 @@ defmodule ElixrTest do
...
+    assert 1 == id
+    assert ~D[2020-01-01] == dat
+  end
 
   # test "should be able to insert a JSON field and read it back", state do
   #   pid = state[:xt]
   #   json_field = %{"foo" => "bar"}
-  #   Postgrex.query!(pid, "INSERT INTO foo8 (_id, json) VALUES (1, '{\"foo\": \"bar\"}'::JSON)", [])
-  #   result = Postgrex.query!(pid, "SELECT * FROM foo8", [])
+  #   Postgrex.query!(pid, "INSERT INTO foo9 (_id, json) VALUES (1, '{\"foo\": \"bar\"}'::JSON)", [])
```
Comment (FiV0) [inline lang/elixr/test/elixr_test.exs:93]: Try to avoid these changes. They are o/w just distracting to the reviewer.

**Review by FiV0 — APPROVED** [review]
LGTM

File: core/src/main/clojure/xtdb/expression.clj [hunk 2]
```diff
@@ -240,6 +240,10 @@
 (defmethod codegen-cast [:num :utf8] [_]
   {:return-type :utf8, :->call-code #(do `(resolve-utf8-buf (str ~@%)))})
 
+(defmethod codegen-cast [:utf8 :uuid] [_]
+  {:return-type :uuid, :->call-code #(do `(util/uuid->byte-buffer
```
Comment (FiV0) [inline core/src/main/clojure/xtdb/expression.clj:244]: @mpisanko Was just relooking at this. I think you wan to return a UUID here.

Comment (mpisanko) [reply core/src/main/clojure/xtdb/expression.clj:244]: looks like that - but having raised the inner form - a test fails (https://github.com/xtdb/xtdb/blob/main/src/test/clojure/xtdb/pgwire_test.clj#L1726) - so i guess it's allright - or the test is wrong?

Comment (FiV0) [reply core/src/main/clojure/xtdb/expression.clj:244]: Yes, forget about this comment. I think we are dealing with multiple types (including UUID) in byte buffer form in the EE.

**Pull request merged**